hs_core/views/__init__.py

```diff
@@ -208,6 +211,15 @@ def update_metadata_element(request, shortkey, element_name, element_id, *args,
 
     return HttpResponseRedirect(request.META['HTTP_REFERER'])
 
+@api_view(['GET'])
+def file_download_url_mapper(request, shortkey, filename):
+    """ maps the file URIs in resourcemap document to django_irods download view function"""
+
+    authorize(request, shortkey, view=True, edit=True, full=True, superuser=True)
```

Comment:
@hyi Not sure what you mean by authorize() is not processed here. Here the purpose of calling authorize() is to raise the necessary exception if the authorization fails. Functionally this call is not needed here as the irods download view function calls the same function.However, checking the authorization/permission as the first thing makes it consistent with other rest api view functions. I agree that we should not be calling the same function twice.  

The api_view decorator is needed for the url mapper view function for client to be able to pass authentication information when accessing this view function. Without the decorator django/REST framework seems to be setting the request object's user property  to anonymous user even when the client sends the user authentication information in its request.